app/models/goal.py

```diff
@@ -3,3 +3,19 @@
 
 class Goal(db.Model):
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    
+    tasks = db.relationship("Task", back_populates="goal", lazy=True)
```

Comment:
Flawless connection between your Goal and Task models -- kudos for remembering SQLAlchemy's `lazy` parameter!